src/core/eme/session.ts

```diff
@@ -314,174 +304,156 @@ function createSessionAndKeyRequest(
     })
     .mapTo(createSessionEvent("generated-request", session, { initData, initDataType }));
 
-  return Observable.merge(sessionEvents, generateRequest);
+  return generateRequest;
 }
 
 /**
- * Create a session, if it fails due to a session.generateRequest error, retry
- * in certain cases.
- * @param {MediaKeys} mediaKeys
- * @param {Object} keySystem
- * @param {string} sessionType - Either "persistent-license" or "temporary"
- * @param {string} initDataType
- * @param {UInt8Array} initData
- * @param {Subject} errorStream
+ * React to each "encrypted" events.
+ * @param {MediaEncryptedEvent} encryptedEvent
+ * @param {Object} mediaKeysInfos
+ * @param {HTMLMediaElement} video
+ * @param {Subject} ErrorStream
  * @returns {Observable}
  */
-function createSessionAndKeyRequestWithRetry(
-  mediaKeys: IMockMediaKeys|MediaKeys,
-  keySystem: IKeySystemOption,
-  sessionType: MediaKeySessionType,
-  initDataType: string,
-  initData: Uint8Array,
-  errorStream: ErrorStream
-): Observable<Event|ISessionEvent> {
-  return createSessionAndKeyRequest(
-    mediaKeys,
-    keySystem,
-    sessionType,
-    initDataType,
-    initData,
-    errorStream
-  )
-    .catch((error) => {
-      if (error.code !== ErrorCodes.KEY_GENERATE_REQUEST_ERROR) {
-        throw error;
+function generateRequestOnSession(
```

Comment:
So this function basically does what generateKeyRequest but with different arguments + a supplementary check.

From the naming, it would seem that this one needs the session in argument and the other does not, but that's not even the case.

Reply:
Also the JSDOC should be updated